File: 백준/Silver/2331. 반복수열/반복수열.cs
using System;

class Program
{
    static int[] check = new int[1000000];
    static int pow(int x,int p)
    {
        int ans = 1;
        for(int i=0;i<p;i++)
        {
            ans = ans*x;
        }
        return ans;
    }
    static int next(int a,int p)
    {
        int ans = 0;
        while(a>0)
        {
            ans += pow(a%10,p);
            a/=10;
        }
        return ans;
    }
    static int length(int a,int p,int cnt)
    {
        if(check[a] != 0) return check[a]-1;
        check[a] = cnt;
        int b = next(a,p);
        return length(b,p,cnt+1);
    }
    static void Main() {
        string[] str = Console.ReadLine().Split(' ');
        int a = int.Parse(str[0]);
        int p = int.Parse(str[1]);
        
        Console.Write(length(a,p,1));
    }
}

File: 백준/Silver/2331. 반복수열/반복수열.py
check = [0] * 1000000

def pow(x,p):
    ans = 1
    for i in range(0,p):
        ans = ans*x
    return ans

def next(a,p):
    ans = 0;
    while a>0:
        ans+=pow(a%10,p)
        a=int(a/10)
    return ans
    
def length(a,p,cnt):
    if check[a] != 0 : 
        return check[a]-1
    check[a] = cnt
    b = next(a,p)
    return length(b,p,cnt+1)

a,p = map(int,input().split())

print(length(a,p,1))